dot/core/messages_test.go

```diff
@@ -114,9 +114,17 @@ func TestServiceHandleTransactionMessage(t *testing.T) {
 	testEmptyHeader := types.NewEmptyHeader()
 	testExtrinsic := []types.Extrinsic{{1, 2, 3}}
 
-	runtimeMock := new(mocksruntime.Instance)
-	runtimeMock2 := new(mocksruntime.Instance)
-	runtimeMock3 := new(mocksruntime.Instance)
+	ctrl := gomock.NewController(t)
+	runtimeMock := NewMockRuntimeInstance(ctrl)
+	runtimeMock2 := NewMockRuntimeInstance(ctrl)
+	runtimeMock3 := NewMockRuntimeInstance(ctrl)
```

Comment:
nit I don't think we actually need those, since there is no expectations on them. You can probably remove them.

Reply:
They are returned in the GetRuntime call

Reply:
You can return `nil` instead, since no method is called on them, it won't crash.

Reply:
Methods are actually called on them. They are defined below in the test

Reply:
Oh yeah my bad. Although ideally these should be defined in the test case locally and not in the parent test. But out of scope really, let's move on!